Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift

```diff
@@ -161,5 +161,34 @@ public extension Array where Element: Equatable {
             }
         }
     }
+}
 
+extension Array where Element: Hashable {
+    
+    public func filtered<E: Equatable>(duplication path: KeyPath<Element, E>) -> [Element] {
+        return reduce(into: [Element]()) { (result, e) in
+            let contains = result.contains { $0[keyPath: path] == e[keyPath: path] }
+            result += contains ? [] : [e]
+        }
+    }
+    
+    public func filtered<E: Equatable>(duplication closure: (Element) throws -> E) rethrows -> [Element] {
```

Comment:
Try to avoid variable names like `array`, `string`, `closure`, etc. I think here a good name is `property` because you're getting a calculated property of the element.
```suggestion
    public func filtered<E: Equatable>(by property: (Element) throws -> E) rethrows -> [Element] {
```

Reply:
Why not reduce into a dictionary keyed by keypath value to avoid the expense of contains?